Formula/cquery.rb

```diff
@@ -13,10 +13,15 @@ class Cquery < Formula
     sha256 "9692d28682c009a9bec583ff20aa16ce5c790975375755a5a48a28ec3e3953c4" => :el_capitan
   end
 
+  depends_on :macos => :sierra
```

Comment:
Is it a documented requirement? Otherwise it's most likely a bug, which needs to be reported and fixed.

Reply:
```
error: 'shared_timed_mutex' is unavailable: introduced in macOS 10.12
  std::shared_timed_mutex dependency_mutex_;
```

This is not a bug. It uses some recent C++ std libs which is not available in macOS 10.11 or below.

Reply:
This is more an Apple's fault. 

Ref: https://stackoverflow.com/questions/34533764/shared-timed-mutex-not-available-on-os-x-10-11-2

Reply:
Would be nice to add a small comment above this line explaining that so it doesn't become a mystery over time, but  I'm okay with the change, personally.

Reply:
Added.